fish-rust/src/wutil/wcstod.rs

```diff
@@ -0,0 +1,444 @@
+use super::errors::Error;
+use crate::wchar::IntoCharIter;
+use fast_float::parse_partial_iter;
+
+/// Parses a 64-bit floating point number.
+///
+/// Leading whitespace and trailing characters are ignored. If the input
+/// string does not contain a valid floating point number (where e.g.
+/// `"."` is seen as a valid floating point number), `None` is returned.
+/// Otherwise the parsed floating point number is returned.
+///
+/// The `decimal_sep` parameter is used to specify the decimal separator.
+/// '.' is a normal default.
+///
+/// The `consumed` parameter is used to return the number of characters
+/// consumed, similar to the "end" parameter to strtod.
+/// This is only meaningful if parsing succeeds.
+///
+/// Error::Overflow is returned if the value is too large in magnitude.
+pub fn wcstod<Chars>(input: Chars, decimal_sep: char, consumed: &mut usize) -> Result<f64, Error>
```

Comment:
`&mut` arguments are much less ergonomic than just returning multiple values - for `wcstod` there's also no risk of mixing up the two return values:

```suggestion
pub fn wcstod<Chars>(input: Chars, decimal_sep: char) -> Result<(f64, usize), Error>
```